docs/source/upgrade/automation.rst

```diff
@@ -0,0 +1,167 @@
+.. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _upgrade-automation:
+
+Automating the Upgrade Process
+##############################
+
+In this section, we are going to automate the upgrade of our coin
+process using :doc:`DAML Script</daml-script/index>` and :doc:`DAML
+Triggers </triggers/index>`. Note that automation for upgrades just
+like the upgrade models is specific to an individual
+application. Nevertheless, we have found that the pattern shown here
+occurs frequently.
+
+Structuring the Upgrade
+=======================
+
+There are three kinds of actions performed during the upgrade:
+
+#. Alice creates ``UpgradeCoinProposal`` contracts. We assume here,
+   that Alice wants to upgrade all ``Coin`` contracts she has
+   issued. Since the ``UpgradeCoinProposal`` proposal is specific to
+   each owner, Alice has to create one ``UpgradeCoinProposal`` per
+   owner. There can be potentially many owners but this step only has
+   to be performed once assuming Alice will not issue more ``Coin``
+   contracts after this point.
+#. Bob and other owners accept the ``UpgradeCoinProposal``. To keep
+   this example simple, we assume that there are only coins issued by
+   Alice. Therefore, each owner has to accept at most one proposal.
+#. As owners accept upgrade proposals, Alice has to upgrade each
+   coin. This means that she has to execute the upgrade choice once
+   for each coin. Owners will not all accept the upgrade at the same
+   time and some might never accept it. Therefore, this should be a
+   long-running process that upgrades all coins of a given owner as
+   soon as they accept the upgrade.
+
+Given those constraints, we are going to use the following tools for
+the upgrade:
+
+#. A DAML script that will be executed once by Alice and creates an
+   ``UpgradeCoinProposal`` contract for each owner.
+#. Navigator to accept the ``UpgradeCoinProposal`` as Bob. While we
+   could also use a DAML script to accept the proposal, this step will
+   often be exposed as part of a web UI so doing it interactively in
+   Navigator resembles that workflow more closely.
+#. A long-running DAML trigger that upgrades all ``Coin`` contracts
+   for which there is a corresponding ``UpgradeCoinAgreeement``.
```

Comment:
```suggestion
   for which there is a corresponding ``UpgradeCoinAgreement``.
```

Reply:
It took me 3 times until I spotted the difference :slightly_smiling_face: